network/iptables_test.go

```diff
@@ -37,6 +37,12 @@ type MockIPTables struct {
 
 func (mock *MockIPTables) ruleIndex(table string, chain string, rulespec []string) int {
 	for i, rule := range mock.rules {
+		if len(rulespec) == 0 {
+			if rule.table == table && rule.chain == chain {
```

Comment:
do this make the test faster?

Reply:
No, For the mock clear chain to work, it needs to match on only table and chain, ignoring the rulespec. So I wrote it as the passed rulespec is empty, it will only match table/chain, otherwise it will look for an exact match of table, chain, rulespec.

Reply:
I added a comment to hopefully answer why this is here.